Frameworks/Foundation/NSURLConnection.mm

```diff
@@ -19,168 +20,289 @@
+        if (!protocol) {
+            TraceError(TAG, L"[NSURLConnection start] could not create a NSURLProtocol for its NSURLRequest.");
+            return;
+        }
+        _protocol = protocol;
+
+        if (_delegateQueue.get()) {
+            [_delegateQueue addOperation:[NSBlockOperation blockOperationWithBlock:^void() {
+                                [_protocol startLoading];
+                            }]];
+
+        } else if (_scheduledRunLoops.size() == 1) {
+            // TODO #: Cannot currently schedule on more than one runloop
+            for (auto pair : _scheduledRunLoops) {
+                [pair.first performSelector:@selector(startLoading) target:_protocol argument:nil order:0 modes:@[ pair.second.get() ]];
+            }
+
+        } else {
+            // Run on the current run loop in the default mode
+            [[NSRunLoop currentRunLoop] performSelector:@selector(startLoading)
+                                                 target:_protocol
+                                               argument:nil
+                                                  order:0
+                                                  modes:@[ NSDefaultRunLoopMode ]];
+        }
     }
+}
 
-    [_protocol startLoading];
+/**
+ @Status Caveat
+ @Notes  TODO #: Cannot currently schedule on more than one runloop
+                 Callbacks do not currently occur on scheduled runloop
+*/
+- (void)scheduleInRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
+    @synchronized(self) {
+        if (_delegateQueue.get()) {
+            [NSException raise:NSInternalInconsistencyException
+                        format:@"A connection cannot be scheduled with both an operation queue and a run loop."];
+        }
+
+        if (_scheduledRunLoops.size() >= 1) {
+            TraceError(TAG, L"NSURLConnection: Scheduling on more than one run loop is currently not supported.");
+            return;
+        }
+
+        StrongId<NSRunLoop> retainedLoop = aRunLoop;
+        StrongId<NSString> retainedMode = mode;
+        _scheduledRunLoops.push_back(std::make_pair(retainedLoop, retainedMode));
+    }
+}
+
+/**
+ @Status Caveat
+ @Notes  TODO #: Callbacks do not currently occur on delegate queue
+*/
+- (void)setDelegateQueue:(NSOperationQueue*)queue {
+    @synchronized(self) {
+        if (_scheduledRunLoops.size() > 0) {
+            [NSException raise:NSInternalInconsistencyException
+                        format:@"A connection cannot be scheduled with both an operation queue and a run loop."];
+        }
+
+        _delegateQueue = queue;
+    }
 }
 
 /**
  @Status Interoperable
 */
-- (void)scheduleInRunLoop:(id)runLoop forMode:(id)mode {
-    _scheduled = YES;
+- (void)unscheduleFromRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
+    @synchronized(self) {
+        for (auto it = _scheduledRunLoops.begin(); it != _scheduledRunLoops.end(); ++it) {
```

Comment:
nit: shouldn't modify a collection while iterating over it, even if it's breaking out of it at the end.  Would prefer std::find() with an erase if != end()